### ISSUE TITLE
Default serialization format for conjunctive graph is turtle, but turtle cannot handle quads  (AFAIK)

### ISSUE THREAD

**Issue by aucampia** [RDFLib/rdflib#1674]
I believe turtle does not handle quads, but our default serialization format for ConjunctiveGraph is turtle as turtle does not override the default for Graph, I guess we should probably change this, question is to what.

I think TriG would make most sense, but the other option is n3.

**Comment by niklasl** [comment]
I agree that TriG is best. [TriG](https://www.w3.org/TR/trig/) is explicitly a superset of Turtle for supporting multiple named graphs (RDF datasets). [Notation 3](https://www.w3.org/TeamSubmission/n3/) is not a W3C REC, and it extends things with more complex structures (variables, graphs as formulas ("complex values") etc.). It is not (AFAIK) a superset of TriG (if it was, it could make sense).

**Comment by aucampia** [comment]
@WhiteGobo made a PR to fix this, but it occurs to me now that fixing this is potentially a breaking change.

There may be some mitigating factors, in that serializing a `ConjunctiveGraph` that contains quads outside the default graph using `turtle` generates output, but the output is wrong. And furthermore, serializing a `ConjunctiveGraph` that only had triples in the default graph as `trig` should result in valid turtle output.

Given these two conditions hold, I think it is okay to include a fix in version 6; otherwise this has to be merged in a way that only has an effect on version 7 onwards[[ref](https://github.com/RDFLib/rdflib/pull/2108)].

CC: @RDFLib/core-reviewers

**Comment by aucampia** [comment]
This issue is somewhat related:

- https://github.com/RDFLib/rdflib/issues/2393

**Comment by aucampia** [comment]
Next release will be 7 and we will generally be more open to breaking changes going forward.

**Comment by nicholascar** [comment]
Closing as we have rolled past this with new major eversion realses
